api/src/tailer/mod.rs

```diff
@@ -0,0 +1,347 @@
+use std::process::Stdio;
+use std::sync::Arc;
+
+use bytes::{Buf, BytesMut};
+use combine::parser::byte::crlf;
+use combine::{
+    error::{ParseError, StreamError},
+    none_of,
+    parser::{
+        combinator::{any_partial_state, AnyPartialState},
+        range::recognize,
+    },
+    skip_many,
+    stream::{easy, PartialStream, RangeStream, StreamErrorFor},
+    Parser,
+};
+use futures::{Stream, StreamExt};
+use tokio::io::{AsyncBufReadExt, BufReader};
+use tokio::process::ChildStdin;
+use tokio::sync::Mutex;
+use tokio_util::codec::{Decoder, FramedRead};
+use tracing::{error, info, trace, warn};
+#[cfg(windows)]
+use win32job::Job;
+
+use http::types::body::LineBuilder;
+
+use crate::tailer::error::TailerError;
+
+mod error;
+
+pub struct TailerApiDecoder {
+    state: AnyPartialState,
+    _stdin: ChildStdin,
+    #[cfg(windows)]
+    _job: Job,
+}
+
+type TailerRecord = String;
+
+// The actual parser for the Tailer API format
```

Comment:
I know we've had issues with encoding in the past, is this inline with current flows?

Reply:
correct. this tailer api is using etf8 encoding, same as JSON payload.